### PR TITLE
Improve performance of python console output

### LINKED ISSUE #1139: Slow performance of Python Console
The Python console output is very slow. A hotspot analysis shows, that the foreach loops in the syntax highlighter are causing a lot of runtime overhead due to unecessary copies.

### PR DESCRIPTION
**Pull request by rainman110**
## Description
Just a small fix to avoid copies in for loops, which turned out to be a hot spot in the syntax highlighter.

This is just a part of the performance fix. The rest need to be done in the python module itself.

Fixes #1139

## How Has This Been Tested?

Manually and via Performance Analysis in Visual Studio.  Before, the for loop took a significant amount of the computing time.

## Checklist:

- [ ] A test for the new functionality was added (if applicable).
- [ ] All tests run without failure.
- [ ] The changelog has been extended, if this MR contains important changes from the users's point of view.
- [ ] The new code complies with the GTlab's style guide.
- [ ] New interface methods / functions are exported via `EXPORT`. Non-interface functions are NOT exported.
- [ ] The number of code quality warnings is not increasing.

### CODE REVIEW

**Comment by mariusalexander** [comment]
have you tried to use a range based for loop instead (together with qAsConst) )? foreach does a copy of the container, which for Qt types is fast but sthis might be faster still. especially if you do not modify the range

**Comment by rainman110** [comment]
> have you tried to use a range based for loop instead (together with qAsConst) )? foreach does a copy of the container, which for Qt types is fast but sthis might be faster still. especially if you do not modify the range

No. It will probably do not harm to add qAsConst.

**Comment by rainman110** [comment]
@mariusalexander I added qAsConst as suggested

File: src/gui/tools/gt_pyhighlighter.cpp [hunk 1]
```diff
@@ -112,7 +112,7 @@ GtPyHighlighter::initializeRules()
 void
 GtPyHighlighter::highlightBlock(const QString& text)
 {
-    foreach (GtHighlightingRule currRule, rules)
+    foreach (const GtHighlightingRule& currRule, qAsConst(rules))
```
Comment (mariusalexander) [inline src/gui/tools/gt_pyhighlighter.cpp:115]:
```suggestion
    for (const GtHighlightingRule& currRule : qAsConst(rules))
```

File: src/gui/tools/gt_pyhighlighter.cpp [hunk 2]
```diff
@@ -43,19 +43,19 @@ GtPyHighlighter::GtPyHighlighter(QTextDocument* parent) :
...
 
-    foreach (QString currOperator, operators)
+    foreach (const QString& currOperator, qAsConst(operators))
     {
         rules.append(GtHighlightingRule(QString("%1").arg(currOperator), 0,
                                         basicStyles.value("operator")));
     }
 
-    foreach (QString currBrace, braces)
+    foreach (const QString& currBrace, qAsConst(braces))
```
Comment (mariusalexander) [inline src/gui/tools/gt_pyhighlighter.cpp:58]:
```suggestion
    for (const QString& currBrace : qAsConst(braces))
```

File: src/gui/tools/gt_pyhighlighter.cpp [hunk 3]
```diff
@@ -43,19 +43,19 @@ GtPyHighlighter::GtPyHighlighter(QTextDocument* parent) :
...
 {
-    foreach (QString currKeyword, keywords)
+    foreach (const QString& currKeyword, qAsConst(keywords))
     {
         rules.append(GtHighlightingRule(QString("\\b%1\\b").arg(currKeyword), 0,
                                         basicStyles.value("keyword")));
     }
 
-    foreach (QString currOperator, operators)
+    foreach (const QString& currOperator, qAsConst(operators))
```
Comment (mariusalexander) [inline src/gui/tools/gt_pyhighlighter.cpp:52]:
```suggestion
    for (const QString& currOperator : qAsConst(operators))
```

File: src/gui/tools/gt_pyhighlighter.cpp [hunk 4]
```diff
@@ -43,19 +43,19 @@ GtPyHighlighter::GtPyHighlighter(QTextDocument* parent) :
 void
 GtPyHighlighter::initializeRules()
 {
-    foreach (QString currKeyword, keywords)
+    foreach (const QString& currKeyword, qAsConst(keywords))
```
Comment (mariusalexander) [inline src/gui/tools/gt_pyhighlighter.cpp:46]:
```suggestion
    for (const QString& currKeyword : qAsConst(keywords))
```

**Review by mariusalexander — COMMENTED** [review]
There was a misunderstanding. I meant the range based for loops ( for ( ... : ...) instead of foreach).

**Review by mariusalexander — APPROVED** [review]

**Pull request merged**